lib/buffer.js

```diff
@@ -351,7 +352,8 @@ function base64ByteLength(str, bytes) {
 
 function byteLength(string, encoding) {
   if (typeof string !== 'string') {
-    if (ArrayBuffer.isView(string) || isArrayBuffer(string))
+    if (ArrayBuffer.isView(string) || isArrayBuffer(string)
+        || isSharedArrayBuffer(string))
```

Comment:
Also, I think curly braces should be added too when converting a single line conditional to a multi-line conditional.